pkg/roachprod/roachprod.go

```diff
@@ -838,18 +836,36 @@ func updatePrometheusTargets(
 				l.Errorf("error getting the port for node %d: %v", index, err)
 				return
 			}
-			nodeInfo := fmt.Sprintf("%s:%d", v.PrivateIP, desc.Port)
+			nodeIP := v.PrivateIP
+			if reachability == promhelperclient.Public {
+				nodeIP = v.PublicIP
+			}
+			nodeInfo := fmt.Sprintf("%s:%d", nodeIP, desc.Port)
 			nodeIPPortsMutex.Lock()
 			// ensure atomicity in map update
-			nodeIPPorts[index] = &promhelperclient.NodeInfo{Target: nodeInfo, CustomLabels: createLabels(v)}
+			if _, ok := nodeIPPorts[index]; !ok {
+				nodeIPPorts[index] = []*promhelperclient.NodeInfo{
+					{
+						Target:       fmt.Sprintf("%s:%d", nodeIP, vm.NodeExporterPort),
+						CustomLabels: createLabels(v, "node_exporter", !c.Secure),
+					},
+				}
+			}
+			nodeIPPorts[index] = append(
+				nodeIPPorts[index],
+				&promhelperclient.NodeInfo{
+					Target:       nodeInfo,
+					CustomLabels: createLabels(v, "cockroachdb", !c.Secure),
+				},
+			)
 			nodeIPPortsMutex.Unlock()
 		}(int(node), c.VMs[node-1])
 
 	}
 	wg.Wait()
 	if len(nodeIPPorts) > 0 {
 		if err := cl.UpdatePrometheusTargets(ctx,
-			c.Name, false, nodeIPPorts, !c.Secure, l); err != nil {
+			c.Name, false, nodeIPPorts, false, l); err != nil {
```

Comment:
question: if secure/insecure targets is now managed through the labels, why do we still need to keep the argument?